spectroscopy/code_src/spitzer_functions.py

```diff
@@ -42,65 +42,66 @@ def SpitzerIRS_get_spec(sample_table, search_radius_arcsec, COMBINESPEC):
                                 spatial="Cone", radius=search_radius_arcsec * u.arcsec)
         print("Number of entries found: {}".format(len(tab)))
 
-        if len(tab) > 0:  # found a source
-
-            # If multiple entries are found, pick the closest.
-            # Or should we take the average instead??
-            if len(tab) > 0:
-                print("More than 1 entry found", end="")
-                if not COMBINESPEC:
-                    print(" - pick the closest")
-                    sep = [search_coords.separation(SkyCoord(tt["ra"], tt["dec"], unit=u.deg, frame='icrs')).to(
-                        u.arcsecond).value for tt in tab]
-                    id_min = np.where(sep == np.nanmin(sep))[0]
-                    tab_final = tab[id_min]
-                else:
-                    print(" - Combine spectra")
-                    tab_final = tab.copy()
+        if len(tab) == 0:
+            continue
+
+        # If multiple entries are found, pick the closest.
+        # Or should we take the average instead??
+        if len(tab) > 0:
```

Comment:
This code block (lines 50-62) is weird in two ways:
- It seems to think the condition was `if len(tab) > 1:` instead of `if len(tab) > 0:` (which would make more sense given that the `== 0` case was already handled).
- The resulting variable `tab_final` is not used. Maybe the following line of code was supposed to use it instead of `tab`?

The diff in this PR makes it difficult to look at, and I'd rather not make substantive changes in this PR anyway. Opening an issue to be addressed separately.